### PR TITLE
Adding init stage

### PR DESCRIPTION
**Pull request by alessandrothea**
See #48

### CODE REVIEW

**Comment by alessandrothea** [comment]
Note: there are rebase conflicts. To be addressed soon...

**Comment by alessandrothea** [comment]
Conflicts fixed.

**Comment by alessandrothea** [comment]
P.S.: Please merge #44 first

File: unittest/FanOutDAQModule_test.cxx [hunk 1]
```diff
@@ -51,8 +51,9 @@ BOOST_AUTO_TEST_CASE(Configure)
 {
   dunedaq::appfwk::FanOutDAQModule<dunedaq::appfwk::NonCopyableType> foum("test");
 
-  auto config = R"({"input": "input"})"_json;
-  foum.configure(config);
+  auto config = R"({"input": "input", "fanout_mode": "RoundRobin", "outputs": []})"_json;
+  foum.set_config(config);
```
Comment (mroda88) [inline unittest/FanOutDAQModule_test.cxx:55]: we should have here the same structure we have in the DAQApplication. simply call `do_init`.

File: unittest/FanOutDAQModule_test.cxx [hunk 2]
```diff
@@ -68,7 +69,8 @@ BOOST_AUTO_TEST_CASE(NonCopyableTypeTest)
                     "fanout_mode": "RoundRobin"
         }
     )"_json;
-  foum.configure(module_config);
+  foum.set_config(module_config);
+  foum.init();
```
Comment (mroda88) [inline unittest/FanOutDAQModule_test.cxx:73]: same here

File: apps/daq_application.cxx [hunk 3]
```diff
@@ -35,7 +35,7 @@ class daq_application_constructor : public GraphConstructor
   /**
    * @brief Constructor for the daq_application_ModuleList
    * @param config_json Configuration file to be used to create the DAQModule
-   * graph
+   * graph`
```
Comment (mroda88) [inline apps/daq_application.cxx:38]: is this ` intentional?

**Pull request merged**